packages/consumption/src/modules/requests/itemProcessors/readAttribute/ReadAttributeRequestItemProcessor.ts

```diff
@@ -1,49 +1,172 @@
+                    return ValidationResult.error(
+                        CoreErrors.requests.attributeQueryMismatch(
+                            "The provided IdentityAttribute is a shared copy of a RepositoryAttribute. You can only share RepositoryAttributes."
+                        )
+                    );
+                }
+
+                const ownSharedIdentityAttributeVersions = await this.consumptionController.attributes.getSharedVersionsOfRepositoryAttribute(
+                    foundLocalAttribute.id,
+                    [requestInfo.peer],
+                    false
+                );
+                const sourceAttributeIdsOfOwnSharedIdentityAttributeVersions = ownSharedIdentityAttributeVersions.map((ownSharedIdentityAttribute) =>
+                    ownSharedIdentityAttribute.shareInfo?.sourceAttribute?.toString()
+                );
+
+                let repositoryAttribute = foundLocalAttribute;
+                let i = 0;
+                while (repositoryAttribute.succeededBy !== undefined && i < 1000) {
+                    const successor = await this.consumptionController.attributes.getLocalAttribute(repositoryAttribute.succeededBy);
+                    if (!successor) {
+                        throw TransportCoreErrors.general.recordNotFound(LocalAttribute, repositoryAttribute.succeededBy.toString());
+                    }
+                    if (sourceAttributeIdsOfOwnSharedIdentityAttributeVersions.includes(successor.id.toString())) {
+                        return ValidationResult.error(
+                            CoreErrors.requests.attributeQueryMismatch(
+                                `The provided IdentityAttribute is outdated. You have already shared the Successor '${successor.id.toString()}' of it.`
+                            )
+                        );
+                    }
+                    repositoryAttribute = successor;
+                    i++;
+                }
+            }
+
+            if (_requestItem.query instanceof ThirdPartyRelationshipAttributeQuery && attribute instanceof RelationshipAttribute) {
+                if (!foundLocalAttribute.isShared()) {
+                    throw new Error("this should never happen");
+                }
+
+                if (foundLocalAttribute.shareInfo.sourceAttribute !== undefined) {
+                    return ValidationResult.error(
+                        CoreErrors.requests.attributeQueryMismatch(
+                            "When responding to a ThirdPartyRelationshipAttributeQuery, only RelationshipAttributes that are not a copy of a sourceAttribute may be provided."
+                        )
+                    );
+                }
+
+                const queriedThirdParties = _requestItem.query.thirdParty.map((aThirdParty) => aThirdParty.toString());
+
+                if (
+                    (this.accountController.identity.isMe(attribute.owner) || queriedThirdParties.includes(attribute.owner.toString())) &&
```

Comment:
I think the line `this.accountController.identity.isMe(attribute.owner) || queriedThirdParties.includes(attribute.owner.toString())` is not necessary, because these things are already checked in the `validateAttributeMatchesWithQuery` function called below. I have added them anyway to optimise the order of the errors thrown, as you should first be informed about an incorrect input for the `owner` compared to this edge case here (which usually also occurs if the `owner` was specified incorrectly), because the simple error message for the `owner` is more useful then. 

Nevertheless, I understand that this is somewhat inelegant. However, I couldn't find a better way than to increase the context here with further queries so that this edge case error is really only thrown in the edge case, but before that a more frequent and easier to resolve error is thrown.

Reply:
I find it hard to comprehend the condition of this if-statement. Can you explain in words to me, why `this.accountController.identity.isMe(attribute.owner) || queriedThirdParties.includes(attribute.owner.toString())` is necessary? And if it's not, I think we should dismiss it, as it increases complexity and doesn't change the returned error.

Reply:
Sure! The line `(this.accountController.identity.isMe(attribute.owner) || queriedThirdParties.includes(attribute.owner.toString()))` is intended to ensure that the subsequent edge case error (`"The provided RelationshipAttribute exists in the context of a Relationship with a third party that should not be involved."`) is only thrown if the RelationshipAttribute under consideration is owned by the Recipient of the Request or one of the requested third parties.

Without this line, this edge case error would also be thrown in the case that neither I nor one of the requested third parties is the owner of the RelationshipAttribute. However, this should be avoided because we should throw errors that are as meaningful as possible and edge case errors should only be thrown in actual edge cases.